harvest/harvest/pipelines.py

```diff
@@ -4,10 +4,42 @@
 # See: https://docs.scrapy.org/en/latest/topics/item-pipeline.html
 
 
+import logging
+
+import backoff
+import requests
+
 # useful for handling different item types with a single interface
 from itemadapter import ItemAdapter  # noqa: F401
 
+LOGGER = logging.getLogger(__name__)
+
 
 class HarvestPipeline:
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.RequestException, max_tries=5
+    )
+    def __indico(self, indico_id):
+        response = requests.get(f"https://indico.cern.ch/export/event/{indico_id}.json")
+
+        response.raise_for_status()
+        data = response.json()
+
+        sponsor = None
+        try:
+            sponsor = data["results"][0]["organizer"].split("/\\-")[0].strip()
+        except Exception:
+            LOGGER.error(f"Cannot get sponsor for Indico event: {indico_id}")
+
+        keywords = None
+        try:
+            keywords = data["results"][0]["keywords"]
+        except Exception:
+            LOGGER.error(f"Cannot get keywords for Indico event: {indico_id}")
+        return sponsor, keywords
+
     def process_item(self, item, spider):
+        indico_id = item.get("event_details")
```

Comment:
it will not work :) you have to get the indico_id from the url

Reply:
How about importing the models and then add
`indico_id = Lecture.objects.get(event_details=item["event_details"]).split("/")[4]`?

Reply:
hmm how you are going to import the models? this is a completely different application :) 

`event_details` is a url, the only thing that you have to do is something like:

```python
item.get("event_details", "").split('/')[-1]
```

Reply:
I overcomplicated it. Thanks!